core/src/main/java/com/statful/client/core/sender/BufferedMetricsSender.java

```diff
@@ -316,4 +323,38 @@ private String buildAggregatedUri(final Aggregation aggregation, final Aggregati
 
         return buffersAsList;
     }
+
+    private void scheduleTask(final Runnable runnable) {
+        boolean wasScheduled = false;
+        if (executorService.getQueue().size() < configuration.getMaxWorkerTasksQueueSize()) {
+            Future<?> futureTask = executorService.schedule(runnable, 0, TimeUnit.MILLISECONDS);
+
+            wasScheduled = futuresQueue.offer(futureTask);
+        }
+
+        if (!wasScheduled) {
+            LOGGER.warning("Unable to send metric! The tasks queue is full. Please lower your timeouts or increase the worker thread pool.");
+            cancelAllWaitingTasks();
+        }
+    }
+
+    private void cancelAllWaitingTasks() {
+        helperExecutorService.schedule(tasksKiller(), 0, TimeUnit.MILLISECONDS);
+    }
+
+    private Runnable tasksKiller() {
+        return new Runnable() {
+            @Override
+            public void run() {
+                Collection<Future<?>> tasks = new ArrayList<Future<?>>();
```

Comment:
Hmmm, yeah. I checked `drainTo` implementation and it uses `add` to insert the elements in the collection. But `LinkedList` can potentially perform better adding. So... 👍